docs/api/annotations.md

```diff
@@ -0,0 +1,233 @@
+
+{
+  ...
+}
+```
+
+
+## Create an annotation
+
+### URL
+
+```http
+POST /annotations HTTP/1.1
+```
+
+### Request parameters
+
+Parameters should be posted as JSON body.
+
+Name             | Type     | Description
+-----------------|----------|------------
+`id`             | `string` | **Optional** The unique identier of the annotation. If none is given, one is generated.
+`doneBase`       | `string` | **Optional** The datetime this annotation was marked as done by the base annotator.
+`doneFull`       | `string` | **Optional** The datetime this annotation was marked as done by the full annotator.
+`recording`      | `string` | **Required** The reference to the recording (type
+`FlatRecordingReference` as specified below.
+`assignment`     | `string` | **Required** The unique identifier of the assignment this annotation
+is for.
+`fragments`      | `list`   | **Required** The annotated fragments (type `Fragment` as specified below).
+
+The Recording format (`FlatRecordingReference`).
+
+Parameters should be posted as JSON body.
+
+Name             | Type     | Description
+-----------------|----------|------------
+`tenant`         | `string` | **Required** The unique identier of the tenant.
+`organisation`   | `string` | **Required** The unique identifier of the organisation.
+`recording`      | `string` | **Required** The unique identifier of the recording.
+`challenge`      | `string` | **Required** The unique identifier of the challenge.
+
+
+The Fragment format:
+
+Parameters should be posted as JSON body.
+
+Name     | Type    | Description
+---------|---------|------------
+`start`  | `float` | **Required** The start point of the fragment, in seconds.
+`end`    | `float` | **Required** The end point of the fragment, in seconds.
+`labels` | `list`  | **Required** The labels assigned to this fragment (type `Label` as specified
+below).
+
+
+The Label format:
+
+Parameters should be posted as JSON body.
+
+Name      | Type     | Description
+----------|----------|------------
+`layer`   | `string` | **Required** The specifier for the layer (Words, OutOfPrompt, Error, Noise).
+`label`   | `string` | **Required** The specifier of a particular label within a layer (Toevoeging ZN, Toevoeging Lidwoord, etc).
+`comment` | `string` | **Required** An optional extra value (e.g. when the label is Toevoeging Lidwoord: De, Het, Een).
+
+
+## Update an annotation
+
+### URL
+
+```http
+PUT /annotations/:annotation HTTP/1.1
+```
+
+* `annotation` - **Required** The annotation identifier.
+
+Update one or more properties of an existing annotation.
+
+### Request parameters
+
+XXX
+
+
+### Request
+
+XXX
+
+### Response
+
+XXX
+
+
+
+# Reference Annotations
+
+A reference annotation is an annotation for a specific audio recording, usually generated through
+forced alignment. This reference annotation can be used as a starting point to create human
+annotations based upon the reference annotation that provides a set of aligned fragments.
```

Comment:
Perhaps it is also good to mention that a reference annotation is actually trying to match the prompt, that was expected to be in the audio, against the recording. That leads to annotations on the fragments where something was found (the alignment). At this point the reference annotation has its focus on the prompt, and not on any error or out of prompt classes.

Reply:
I noticed the reference is actually created as soon as a recording is created. I do not know why. I am also not sure if that should be part of the docs here, at the SpeechRecording or not at all. You thoughts?

Reply:
A reference is probably null at the point a recording is created? So I won't bother.

Reply:
I noticed it has some metadata (perhaps that means Null in the DB, did not check).
What is interesting here that it actually explains why you need to PUT in stead of POST (and that is quite important to note).